File: admin-panel/application/src/types/feedback.ts
export interface Feedback {
  id: string
  trip_id: string
  trip_title: string | null
  user_id: string
  user_email: string
  overall_rating: number
  highlights: string | null
  lowlights: string | null
  would_recommend: boolean
  created_at: string
}

File: bagtrip/lib/models/feedback.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'feedback.freezed.dart';
part 'feedback.g.dart';

@freezed
abstract class TripFeedback with _$TripFeedback {
  const factory TripFeedback({
    required String id,
    required String tripId,
    required String userId,
    required int overallRating,
    String? highlights,
    String? lowlights,
    @Default(false) bool wouldRecommend,
    DateTime? createdAt,
  }) = _TripFeedback;

  factory TripFeedback.fromJson(Map<String, dynamic> json) =>
      _$TripFeedbackFromJson(json);
}
